### PR TITLE
Use handleName field consistently in HandleController

### PR DESCRIPTION
**Pull request by peterjanes**
## Description

There are two instances of `handle.name` that should be `handle.handleName` so that the handle map is populated correctly.

## Related issue(s)

- #2035

## Motivation and context

I haven't nailed down the exact conditions, but sometimes when adding `sp-slider`s dynamically from an HTML template only one instance of `sp-slider-handle` will be added because the handle map key is undefined. (I finally tracked down the behaviour when I started getting `First slider handle cannot have attribute min="previous"` after adding `max="next"` to the first `sp-slider-handle` and `min="previous"` to the second.)

## How has this been tested?

I'm afraid I don't have a reduced test case, but I was able to track down the behaviour in my (currently non-public) app. The code does something like this (which doesn't reproduce the issue in isolation, but is fixed by the change in this PR):

```js
const sliders = document.createElement('div');
arr.forEach(parms => {
  const el = sliderTemplate.content.cloneNode(true);
  const slider = el.querySelector('sp-slider');
  slider.setAttribute('label', parms.description);
  slider.querySelector('[name=min]').setAttribute('value', parms.min);
  slider.querySelector('[name=max]').setAttribute('value', parms.max);
  sliders.appendChild(slider);
});
document.body.appendChild(sliders);
```

## Types of changes

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

File: packages/slider/src/HandleController.ts [hunk 1]
```diff
@@ -208,7 +208,7 @@ export class HandleController implements Controller {
         handles.forEach((handle, index) => {
             /* c8 ignore next */
             if (!handle.handleName?.length) {
-                handle.name = `handle${index + 1}`;
+                handle.handleName = `handle${index + 1}`;
```
Comment (Westbrook) [inline packages/slider/src/HandleController.ts:211]: This seems weird in that `handle.handleName` is _only_ a getter for `handle.name`...

Comment (Westbrook) [reply packages/slider/src/HandleController.ts:211]: https://github.com/adobe/spectrum-web-components/blob/d23f735db3cea01c2243e0485448d1598c6a8692/packages/slider/src/SliderHandle.ts#L80-L82

Comment (peterjanes) [reply packages/slider/src/HandleController.ts:211]: Should've said, my "fix" is wrong (in editing it while I was waiting for CLA approval I removed that minor detail) but I'm not sure why the handle's `name` property isn't being initialized properly.

Comment (Westbrook) [reply packages/slider/src/HandleController.ts:211]: Could you share into the issue more info around how you add the DOM to the page, etc.?

As this change isn't right, let's close this PR and discuss finding the source of this in the issue.

**Comment by Westbrook** [comment]
Let's look more for a source before we review a fix.

**Pull request closed without merging**